service/dap/server.go

```diff
@@ -518,42 +565,68 @@ func (s *Server) onLaunchRequest(request *dap.LaunchRequest) {
 
 	s.setLaunchAttachArgs(request)
 
-	var targetArgs []string
-	args, ok := request.Arguments["args"]
-	if ok {
-		argsParsed, ok := args.([]interface{})
-		if !ok {
-			s.sendErrorResponse(request.Request,
-				FailedToLaunch, "Failed to launch",
-				fmt.Sprintf("'args' attribute '%v' in debug configuration is not an array.", args))
-			return
-		}
-		for _, arg := range argsParsed {
-			argParsed, ok := arg.(string)
-			if !ok {
-				s.sendErrorResponse(request.Request,
-					FailedToLaunch, "Failed to launch",
-					fmt.Sprintf("value '%v' in 'args' attribute in debug configuration is not a string.", arg))
-				return
-			}
-			targetArgs = append(targetArgs, argParsed)
-		}
+	targetArgs, ok := s.parseTargetArgs(request, ok)
+	if !ok {
+		return
 	}
 
 	s.config.ProcessArgs = append([]string{program}, targetArgs...)
 	s.config.Debugger.WorkingDir = filepath.Dir(program)
 
 	// Set the WorkingDir for this program to the one specified in the request arguments.
-	wd, ok := request.Arguments["wd"]
-	if ok {
-		wdParsed, ok := wd.(string)
-		if !ok {
+	if !s.setWorkingDir(request) {
+		return
+	}
+
+	if mode == "record" {
+		// Validate parameters and record an rr trace from the compiled binary
+		s.config.Debugger.Backend = "rr"
+
+		// Record only works on rr trace directories
+		traceDirectory, ok := request.Arguments["traceDirectory"].(string)
+		if !ok || traceDirectory == "" {
```

Comment:
No reason to check ok here, when ok is false traceDirectory will always be the empty string.